deploy/deploy_test.go

```diff
@@ -576,13 +690,18 @@ func TestMetalLBSpec(t *testing.T) {
 			}
 			k.PrependWatchReactor("deployments", reaction)
 		},
+		resp: []fexec.Response{
+			{Cmd: "kubectl", Args: []string{"apply", "-f", ""}},
+		},
 		dErr: "dclient error",
 	}, {
 		desc: "valid deployment",
 		m: &MetalLBSpec{
 			IPCount: 20,
 		},
-		execer: exec.NewFakeExecer(nil, nil, nil),
+		resp: []fexec.Response{
```

Comment:
why are some of these going from 3 execs to one? was it just a case of the old test having the incorrect amount?

Reply:
There is only one execution during the test.  When I add {Cmd: "x"}, to the list I get
```
    --- FAIL: TestMetalLBSpec/valid_deployment (0.00s)
        deploy_test.go:892: valid deployment: {Cmd: "kubectl", Args: []string{"apply", "-f", ""}}
        deploy_test.go:960: Done: didn't execute:
            	{Cmd: "x"}
```

So yes, I think the old code just had extra nil's in it.